datasets/covtype.py

```diff
@@ -3,11 +3,30 @@
 
 with safe_import_context() as import_ctx:
     import numpy as np
-    from benchmark_utils import oracles
     from sklearn.datasets import fetch_covtype
-    from sklearn.preprocessing import StandardScaler
     from sklearn.model_selection import train_test_split
-    from benchmark_utils.oracle_utils import convert_array_framework
+    from sklearn.preprocessing import StandardScaler, OneHotEncoder
+
+    import jax
+    import jax.numpy as jnp
+    from jax.nn import logsumexp
+
+    from functools import partial
+
+
+def loss_sample(inner_var_flat, outer_var, x, y):
+    n_classes = y.shape[0]
+    n_features = x.shape[0]
+    inner_var = inner_var_flat.reshape(n_features, n_classes)
+    prod = jnp.dot(x, inner_var)
+    lse = logsumexp(prod)
```

Comment:
This makes it easier to read the code, as we know where the func comes from.
```suggestion
    lse = jax.nn.logsumexp(prod)
```